pkg/cmd/pulumi/config.go

```diff
@@ -917,6 +947,38 @@ func listConfig(ctx context.Context,
 			Headers: []string{"KEY", "VALUE"},
 			Rows:    rows,
 		}, nil)
+
+		if env != nil {
+			_, environ, _, err := cli.PrepareEnvironment(env, &cli.PrepareOptions{
+				Pretend: true,
+				Redact:  true,
+			})
+			if err != nil {
+				return err
+			}
+
+			if len(environ) != 0 {
+				environRows := make([]cmdutil.TableRow, len(environ))
+				for i, kvp := range environ {
+					key, value, _ := strings.Cut(kvp, "=")
+					environRows[i] = cmdutil.TableRow{Columns: []string{key, value}}
+				}
+
+				fmt.Println()
+				printTable(cmdutil.Table{
+					Headers: []string{"ENVIRONMENT VARIABLE", "VALUE"},
```

Comment:
Seems like there could be some confusion over system envvars, but I don't have a great alternative suggestion if this is indeed the terminology.

Reply:
Does ESC actually call these variables or keys/paths?

Reply:
These are actual system environment variables that will be set during `up`/`preview`/etc. using values plucked from the `environmentVariables` key in the stack's environment.

For example, if my stack references an environment that looks like this:

```yaml
# Here's the environment config
values:
    aws:
        creds:
            fn::open::aws-login:
                oidc:
                    duration: 1h
                    roleArn: <role-arn>
                    sessionName: pulumi-session
    environmentVariables:
        AWS_ACCESS_KEY_ID: ${aws.creds.accessKeyId}
        AWS_SECRET_ACCESS_KEY: ${aws.creds.secretAccessKey}
        AWS_SESSION_TOKEN: ${aws.creds.sessionToken}
```

Then we will set the `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_SESSION_TOKEN` environment variables during Pulumi operations. With these changes we would display those environment variables in the output of `pulumi config` as well.

Reply:
Ah, ok.